include/exadg/postprocessor/pointwise_output_generator_base.h

```diff
@@ -0,0 +1,202 @@
+                        std::vector<Number> const & values,
+                        unsigned int const          first_component = 0)
+  {
+    AssertThrow(first_component == 0,
+                dealii::ExcMessage("This function can only be called with first_component==0."));
+
+    componentwise_result = std::move(dealii::Vector<Number>(values.begin(), values.end()));
+    write_component(name, componentwise_result);
+  }
+
+  template<int n_components>
+  void
+  write_scalar_quantity(std::string const &                                          name,
+                        std::vector<dealii::Tensor<1, n_components, Number>> const & values,
+                        unsigned int const first_component = 0)
+  {
+    this->extract_component(componentwise_result, values, first_component);
+    write_component(name, componentwise_result);
+  }
+
+  template<int n_components>
+  void
+  write_vectorial_quantity(std::string const &                                          name,
+                           std::vector<dealii::Tensor<1, n_components, Number>> const & values,
+                           unsigned int const first_component = 0)
+  {
+    for(unsigned int d = 0; d < dim; ++d)
+    {
+      this->extract_component(componentwise_result, values, d + first_component);
+      write_component(name + std::to_string(d), componentwise_result);
+    }
+  }
+
+  template<int n_components>
+  [[nodiscard]] std::vector<
+    typename dealii::FEPointEvaluation<n_components, dim, dim, Number>::value_type>
+  compute_point_values(VectorType const & solution) const
+  {
+    return dealii::VectorTools::point_values<n_components>(*remote_evaluator,
+                                                           *dof_handler,
+                                                           solution);
+  }
+
+private:
+  virtual void
+  do_evaluate(VectorType const & solution) = 0;
+
+  void
+  setup_remote_evaluator();
+
+  void
+  reinit_remote_evaluator();
+
+  void
+  create_hdf5_file();
+
+#ifdef DEAL_II_WITH_HDF5
+  void
+  add_evaluation_points_dataset(dealii::HDF5::Group & group, std::string const & name);
+#endif
+
+  void
+  fill_evaluation_points(std::string const & name);
+
+#ifdef DEAL_II_WITH_HDF5
+  void
+  add_time_dataset(dealii::HDF5::Group & group, std::string const & name);
+#endif
+
+  void
+  write_time(double time);
+
+  void
+  write_component(std::string const & name, dealii::Vector<Number> const & componentwise_result);
+
+  template<int n_components>
+  void
+  extract_component(dealii::Vector<Number> &                                     dst,
+                    std::vector<dealii::Tensor<1, n_components, Number>> const & values,
+                    unsigned int const                                           comp) const
+  {
+    AssertIndexRange(comp, n_components);
+
+    auto iter = dst.begin();
+    for(auto const & val : values)
+    {
+      *iter = val[comp];
+      ++iter;
+    }
+  }
```

Comment:
This functionality is so general that it should appear as a free function at a central place in ExaDG (in directory utilities) in my opinion.

Reply:
[ce6a59b](https://github.com/exadg/exadg/pull/255/commits/ce6a59b5edabc2a6164a6c62e803c748b3fdce23)